readiness/e2e/suite_test.go

```diff
@@ -0,0 +1,470 @@
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1alpha2 "github.com/vmware-tanzu/tanzu-framework/apis/core/v1alpha2"
+)
+
+var (
+	timeout         = 2 * time.Minute
+	pollingInterval = 50 * time.Millisecond
+	testNamespace   = "default"
+)
+
+var _ = Describe("Readiness end to end suite", func() {
+	Context("Readiness with no checks", func() {
+		err := cl.Create(context.TODO(), &v1alpha2.Readiness{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "readiness-without-checks",
+			},
+			Spec: v1alpha2.ReadinessSpec{
+				Checks: []v1alpha2.Check{},
+			},
+		})
+		Expect(err).To(BeNil())
+
+		It("Should reconcile to ready state", func() {
+			Eventually(func() bool {
+				readiness := &v1alpha2.Readiness{}
+				err := cl.Get(context.TODO(), types.NamespacedName{Name: "readiness-without-checks"}, readiness)
+				return err == nil && readiness.Status.Ready == true
+			}).WithTimeout(5 * timeout).WithPolling(pollingInterval).Should(BeTrue())
+
+			deleteReadinessResources()
```

Comment:
Can the cleanup be done in `AfterSuite()` ?

Reply:
Done